books/spiders/products.py

```diff
@@ -0,0 +1,38 @@
+import scrapy
+from scrapy.http import Response
+
+
+class ProductsSpider(scrapy.Spider):
+    name = "products"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> str:
+        for product in response.css(".product_pod"):
+            detail_page = response.urljoin(
+                product.css("h3 > a::attr(href)").get()
+            )
+            yield response.follow(
+                detail_page, callback=self.parse_detail_product
+            )
+        next_page = response.css(".next a::attr(href)").get()
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def parse_detail_product(self, response: Response) -> dict:
+        rating = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}
```

Comment:
you can define it globally as const, but it`s up to u

Reply:
agree